File: src/exemplos/03_Fluxo/02_Repeticao/00-while.c
/**      @file: 00-while.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo do laço de repetição WHILE. */

#include <stdio.h>

#define PICANHA "free(B01)"

void sirva(char* opcao) {
  printf("Servindo %s.\n\n", opcao);
}

int main() {
  char resposta = 'n';

  printf("Bem-vindo ao rodízio!\n");

  while(resposta == 'n' || resposta == 'N') {
    sirva(PICANHA);

    printf("Satisfeito?\n");
    scanf(" %c", &resposta); /* <--  Atualização da      */
  }                          /*      condição de parada. */

  return 0;
}

File: src/exemplos/03_Fluxo/02_Repeticao/00-while.py
#  -*- coding: utf-8 -*-
#    @package: 00-while.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo do laço de repetição WHILE.


PICANHA = 'free(B01)'


def sirva(opcao):
    '''Mostra o que está sendo servido.'''
    print('Servindo %s.' % opcao)


resposta = 'n'

print('Bem-vindo ao rodízio!')

while resposta == 'n' or resposta == 'N':
    sirva(PICANHA)

    # Atualização da condição de parada.
    resposta = input('Satisfeito? ')
